tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/XPathAlt.java

```diff
@@ -0,0 +1,286 @@
+
+        @Schema(description = "Array of possible alt attributes.")
+        public final String[] alt;
+
+        public AltSetResponse(String hexId, Set<String> set) {
+            this.hexId = hexId;
+            this.alt = (set == null) ? new String[0] : set.toArray(new String[0]);
+        }
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Add alt path", description = "Create a new alt path in the given locale")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "404",
+                description = "XPath Hex ID not found",
+                content = @Content(mediaType = "application/json", schema = @Schema(implementation = STError.class))
+            ),
+            @APIResponse(
+                responseCode = "200",
+                description = "Alt path was added",
+                content = @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = AddAltPathRequest.class)
+                )
+            ),
+        }
+    )
+    public Response putByHex(@HeaderParam(Auth.SESSION_HEADER) String sessionString, AddAltPathRequest request) {
+        CookieSession session;
+        try {
+            session = Auth.getSession(sessionString);
+            if (session == null) {
+                return Auth.noSessionResponse();
+            }
+            if (!UserRegistry.userIsTC(session.user)) {
+                return Response.status(403, "Forbidden").build();
+            }
+            session.userDidAction();
+        } catch (Exception e) {
+            return Response.status(500, "An exception occurred").entity(e).build();
+        }
+        CLDRLocale locale = CLDRLocale.getInstance(request.localeId);
+        if (locale == null) {
+            return badLocale(request.localeId);
+        }
+        String xpath = getXPathByHex(request.hexId);
+        if (xpath == null) {
+            return badPath(request.hexId);
+        }
+        String alt = getFromSet(request.alt, locale, xpath);
+        if (alt == null) {
+            return noAlt(request.hexId, request.alt);
+        }
+        return addPath(locale, xpath, alt, session);
+    }
+
+    private String getFromSet(String requestAlt, CLDRLocale locale, String xpath) {
+        Set<String> set = getSet(locale, xpath);
+        if (set == null || !set.contains(requestAlt)) {
+            return null;
+        }
+        for (String alt: set) {
+            if (alt.equals(requestAlt)) {
+                // do not return requestAlt -- even though they're equal, alt differs from
+                // requestAlt in the eyes of Code scanning / CodeQL warning,
+                // "Polynomial regular expression used on uncontrolled data"
+                // -- requestAlt is "uncontrolled data"
+                // -- alt is "controlled data"
+                return alt;
+            }
```

Comment:
The "code scanning results" reported earlier appear to have been fixed by my squashed commit with this new method